resources/web/map/js/trajectory_raw.js

```diff
@@ -131,19 +194,37 @@ new QWebChannel(qt.webChannelTransport, (channel) => {
             startMarker.remove();
             startMarker = null;
         }
+        if (currentMarker) {
+            currentMarker.remove();
+            currentMarker = null;
+        }
     });
 
     chn.recvPos.connect((id, lat, lng, hAcc) => {
-        const pos = [lat, lng],
-            rX = hAcc / 1000;
+        const pos = [lat, lng], rX = hAcc / 1000;
         data[id].features.push(createGeoJsonEllipse(pos, rX, rX));
+        crumbCoords.push(pos);
         if (!map) return;
+        if (!currentMarker) currentMarker = new mapboxgl.Marker().setLngLat(pos).addTo(map);
+        else currentMarker.setLngLat(pos);
         if (!startMarker) {
             startMarker = new mapboxgl.Marker().setLngLat(pos).addTo(map);
             map.panTo(pos);
         } else if (focusCurrent) map.panTo(pos);
         let src = map.getSource(`route${id}`);
-        if (src != null) src.setData(data[id]);
+        if (src) src.setData(data[id]);
+
+        if (map.getSource('breadcrumb')) syncCrumbCoords();
+    })
+
+    chn.protPos.connect((lat, lng, hpl) => {
+        const pos = [lng, lat], rX = hpl / 100_000; // hpl in cm, convert to km
+        if (!map) return;
+        let src = map.getSource(`prot`);
+        if (src) src.setData({
+            type: 'FeatureCollection',
+            features: [createGeoJsonEllipse(pos, rX, rX)]
```

Comment:
I'm not familiar enough with this message type in the sbp protocol but is there ever a case when these values should be reset to 0 or a maximum duration before these need to get reset?

Reply:
hmmmmm oh yeah not quite sure when it should get reset
with the protocol also not sure whether its always sent along with the LLH msg

(but it will just persist if theres no new protection lvl msg, itll be like last seen position with protection lvl)
^^ will have to do more digging into drongo

Reply:
Maybe you could kick off a timer event of some sort and reset to zero after 1sec has elapsed. I imagine we would be doing something similar in production. What are your thoughts @dgburr?

Reply:
ok yea i think it seems to have the same functionality compared to drongo

sbp file in: https://swift-nav.atlassian.net/browse/CPP-915

Reply:
Fair enough, yeah I guess if drongo is reusing llh then it is probably acceptable :shrug:

Reply:
If I understand correctly, there are two questions here:

Q. Should the centre of the EHPE ellipses be the same as the centre of the Protection Level ellipses?  
A: Yes.

Q. When should the Protection Level ellipses disappear?
A: The way the SBP protocol works is that it is necessary to combine data from multiple SBP messages with the same timestamp to determine the complete state at any given moment.  In the source code for Drongo and Woodpecker, I use the term "epoch" to refer to this concept. One epoch can contain 0 or 1 HPL values (depending upon Starling was configured was configured to output MSG_PROTECTION_LEVEL messages).  If a HPL value is present in an epoch, then the PL circle should remain on the screen for as long as the epoch is being displayed.  Typically this would mean that it stays on the screen until all data for the next epoch is received, after which the old ellipse is removed and a new ellipse is drawn.